pkg/unikontainers/unikernels/rumprun.go

```diff
@@ -75,42 +75,51 @@ func (r Rumprun) CommandString() (string, error) {
 	return jsonStr, nil
 }
 
-func newRumprun(data UnikernelParams) (Rumprun, error) {
-	tempBlk := RumprunBlk{
-		Source:     "etfs",
-		Path:       "/dev/ld0a",
-		FsType:     "blk",
-		Mountpoint: "/data",
+func (r *Rumprun) SupportsBlock() bool {
+	return true
+}
+
+func (r *Rumprun) SupportsFS(fsType string) bool {
+	switch fsType {
+	case "ext2":
+		return true
+	default:
+		return false
 	}
+}
+
+func (r *Rumprun) Init(data UnikernelParams) error {
 
 	// if EthDeviceMask is empty, there is no network support
-	if data.EthDeviceMask == "" {
-		return Rumprun{
-			Command: data.CmdLine,
-			Net:     RumprunNet{Mask: ""},
-			Blk:     tempBlk,
-		}, nil
+	if data.EthDeviceMask != "" {
+		// FIXME: in the case of rumprun & k8s, we need to explicitly set the mask
+		// to an inclusive value (eg 1 or 0), as NetBSD complains and does not set the default gw
+		// if it is not reachable from the IP address directly.
+		mask, err := subnetMaskToCIDR(data.EthDeviceMask)
+		if err != nil {
+			return err
+		}
+		r.Net.Interface = "ukvmif0"
+		r.Net.Cloner = "True"
+		r.Net.Type = "inet"
+		r.Net.Method = "static"
+		r.Net.Address = data.EthDeviceIP
+		r.Net.Mask = fmt.Sprintf("%d", mask)
+		r.Net.Gateway = data.EthDeviceGateway
 	}
 
-	// FIXME: in the case of rumprun & k8s, we need to explicitly set the mask
-	// to an inclusive value (eg 1 or 0), as NetBSD complains and does not set the default gw
-	// if it is not reachable from the IP address directly.
-	mask, err := subnetMaskToCIDR(data.EthDeviceMask)
-	if err != nil {
-		return Rumprun{}, err
-	}
-	tempNet := RumprunNet{
-		Interface: "ukvmif0",
-		Cloner:    "True",
-		Type:      "inet",
-		Method:    "static",
-		Address:   data.EthDeviceIP,
-		Mask:      fmt.Sprintf("%d", mask),
-		Gateway:   data.EthDeviceGateway,
-	}
-	return Rumprun{
-		Command: data.CmdLine,
-		Net:     tempNet,
-		Blk:     tempBlk,
-	}, nil
+	r.Blk.Source = "etfs"
+	r.Blk.Path = "/dev/ld0a"
+	r.Blk.FsType = "blk"
+	r.Blk.Mountpoint = "/data"
+
+	r.Command = data.CmdLine
+
+	return nil
+}
+
+func newRumprun() (*Rumprun, error) {
```

Comment:
Perhaps we could change the function signature to omit the returned error, or even remove this function and use new(Rumprun) directly from unikernel.go>New()

Reply:
Yes that is a good point. I will remove the error return. If we need it we can add it later.

Reply:
However, I think we should keep the constructor in the respective unikernel file, because we might want to change it in the future with something that might be specific for a unikernel. Would that make sense?

Reply:
Yes, it could prove useful to have the constructor in place.